backend/corpora/lambdas/api/v1/curation/collections/actions.py

```diff
@@ -18,23 +18,28 @@ def get(visibility: str, token_info: dict, curator: str = None):
     @return: Response
     """
     filters = [DbCollection.tombstone == False]  # noqa
-
-    if visibility == CollectionVisibility.PRIVATE.name and not token_info:
-        raise UnauthorizedError()
-    elif visibility:
+    if visibility:
         filters.append(DbCollection.visibility == getattr(CollectionVisibility, visibility))
+        if visibility == CollectionVisibility.PRIVATE.name:
+            if not token_info:
+                raise ForbiddenHTTPException(detail="Not authorized to query for PRIVATE collection.")
+            else:
+                result = owner_or_allowed(token_info)
+                if result:
+                    filters.append(DbCollection.owner == result)
```

Comment:
to accommodate Super Curator #3 outlined [here](https://github.com/chanzuckerberg/single-cell-data-portal/pull/3089#issuecomment-1213348180), I think the owner filter should only be appended if the user is not a super curator. Maybe turn this else block into an `elif not is_super_curator(token_info)`

Reply:
nvm, owner_or_allowed already returns None for superusers. Maybe could use a rename, but that's out of scope for this ticket

Reply:
I'll leave this open as a non-blocking suggestion in case you do think of a better name, because it isn't clear just reading a call to owner_or_allowed what it actually returns. but admittedly, I can't come up with anything particularly great off the top of my head

Reply:
I change the name to owner_filter and added a comment.